src/particle_filter.cpp

```diff
@@ -1,33 +1,95 @@
 #include "particle_filter.h"
 
 #include <algorithm>
+#include <cmath>
+
+#include <ros/console.h>
 
 // ----------------------------------------------------------------------------------------------------
 
-ParticleFilter::ParticleFilter() : i_current_(0)
+ParticleFilter::ParticleFilter() : min_samples_(0), max_samples_(0), kld_err_(0), kld_z_(0), alpha_slow_(0),
```

Comment:
Here, you could consider initializing one variable per line (though that takes a bit more space)